release/serve_tests/workloads/scalability_1k_replicas.py

```diff
@@ -0,0 +1,161 @@
+        "route_prefix": "/",
+        "runtime_env": {"working_dir": "workloads"},
+        "deployments": [
+            {
+                "name": "Noop",
+                "num_replicas": num_replicas,
+                "ray_actor_options": {"resources": {"worker_resource": 0.01}},
+            }
+        ],
+    }
+    compute_config = ComputeConfig(
+        cloud="anyscale_v2_default_cloud",
+        head_node=HeadNodeConfig(instance_type="m5.8xlarge"),
+        worker_nodes=[
+            WorkerNodeGroupConfig(
+                instance_type="m5.8xlarge",
+                min_nodes=0,
+                max_nodes=50,
+                resources={"worker_resource": 1},
+            ),
+        ],
+    )
+    stages = [
+        LocustStage(
+            duration_s=trial_length,
+            users=50,
+            spawn_rate=10,
+        ),
+        LocustStage(
+            duration_s=trial_length,
+            users=100,
+            spawn_rate=20,
+        ),
+        LocustStage(
+            duration_s=trial_length,
+            users=500,
+            spawn_rate=100,
+        ),
+        LocustStage(
+            duration_s=trial_length,
+            users=1000,
+            spawn_rate=200,
+        ),
+    ]
+
+    with start_service(
+        service_name="scalability-1k-replicas",
+        compute_config=compute_config,
+        applications=[noop_1k_application],
+    ) as service_name:
+        ray.init("auto")
+        status = service.status(name=service_name)
+
+        # Start the locust workload
+        num_locust_workers = int(ray.available_resources()["CPU"]) - 1
+        stats = run_locust_load_test(
+            LocustLoadTestConfig(
+                num_workers=num_locust_workers,
+                host_url=status.query_url,
+                auth_token=status.query_auth_token,
+                data=None,
+                stages=stages,
+            )
+        )
+
+        results_per_stage = [
+            [
+                {
+                    "perf_metric_name": f"stage_{i+1}_p50_latency",
+                    "perf_metric_value": stats.stats_in_stages[i].p50_latency,
+                    "perf_metric_type": "LATENCY",
+                },
+                {
+                    "perf_metric_name": f"stage_{i+1}_p90_latency",
+                    "perf_metric_value": stats.stats_in_stages[i].p90_latency,
+                    "perf_metric_type": "LATENCY",
+                },
+                {
+                    "perf_metric_name": f"stage_{i+1}_p99_latency",
+                    "perf_metric_value": stats.stats_in_stages[i].p99_latency,
+                    "perf_metric_type": "LATENCY",
+                },
+                {
+                    "perf_metric_name": f"stage_{i+1}_rps",
+                    "perf_metric_value": stats.stats_in_stages[i].rps,
+                    "perf_metric_type": "THROUGHPUT",
+                },
+            ]
+            for i in range(len(stages))
+        ]
+        results = {
```

Comment:
is it possible to also collect controller metrics? we currently export some grafana metrics like the loop latency and interval

the controller is likely to be the biggest bottleneck here. we can also run this test a few times manually and see what's missing when making improvements